src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -2094,30 +2099,65 @@ TEST_CASE("temp entry eviction", "[tx][soroban]")
     REQUIRE(!test.isEntryLive("temp", ContractDataDurability::TEMPORARY,
                               ledgerSeq));
 
-    // close one more ledger to trigger the eviction
-    closeLedgerOn(*test.getApp(), 4097, 2, 1, 2016);
+    SECTION("eviction")
+    {
+        // close one more ledger to trigger the eviction
+        closeLedgerOn(*test.getApp(), 4097, 2, 1, 2016);
+
+        {
+            LedgerTxn ltx(test.getApp()->getLedgerTxnRoot());
+            REQUIRE(!ltx.load(lk));
+        }
 
-    XDRInputFileStream in;
-    in.open(metaPath);
-    LedgerCloseMeta lcm;
-    bool evicted = false;
-    while (in.readOne(lcm))
+        XDRInputFileStream in;
+        in.open(metaPath);
+        LedgerCloseMeta lcm;
+        bool evicted = false;
+        while (in.readOne(lcm))
+        {
+            REQUIRE(lcm.v() == 1);
+            if (lcm.v1().ledgerHeader.header.ledgerSeq == 4097)
+            {
+                REQUIRE(lcm.v1().evictedTemporaryLedgerKeys.size() == 2);
+                auto sortedKeys = lcm.v1().evictedTemporaryLedgerKeys;
+                std::sort(sortedKeys.begin(), sortedKeys.end());
+                REQUIRE(sortedKeys[0] == lk);
+                REQUIRE(sortedKeys[1] == getTTLKey(lk));
+                evicted = true;
+            }
+            else
+            {
+                REQUIRE(lcm.v1().evictedTemporaryLedgerKeys.empty());
+            }
+        }
+
+        REQUIRE(evicted);
+    }
+
+    SECTION("Create temp entry with same key as an expired entry on eviction "
+            "ledger")
     {
-        REQUIRE(lcm.v() == 1);
-        if (lcm.v1().ledgerHeader.header.ledgerSeq == 4097)
+        auto tx2 = test.createInvokeTx(resources, funcSym, {keySym, valU64},
```

Comment:
IIUC the intention here is for entry to be evicted, will the section above actually be executed before this? (IIRC sections are exclusive, aren't they?)

Reply:
I'm not sure I understand what you mean. The point of this section is to close the ledger where it was supposed to get evicted, but instead in gets recreated.

Reply:
I see, I thought the intent was to create entry in the ledger following the eviction. This looks fine then.